marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubConfigImpl.java

```diff
@@ -2016,4 +2016,75 @@ public void resetAppConfigs() {
         manageClient = null;
     }
 
+    public void resetHubConfigs() {
+        stagingDbName = null;
```

Comment:
I think reflection would be useful here to lessen the maintenance burden - i.e. any time a new field is added to HubConfigImpl, we have to hope that someone remembers to null it out in this method if necessary. Could reflect on all the fields and null out anything starting with "staging" or "final" or "job" or "modules", and then also null out the handful of other fields not starting with one of those prefixes.

Reply:
@hao1st , I agree with @rjrudin .  HubTestBase.resetProperties() nulls out fields using reflection. You can modify the logic.

```
    public void resetProperties() {
        Field[] fields = HubConfigImpl.class.getDeclaredFields();
        Set<String> s =  Stream.of("hubProject", "environment", "flowManager", 
                "dataHub", "versions", "logger", "objmapper", "projectProperties").collect(Collectors.toSet());

        for(Field f : fields){
            if(! s.contains(f.getName())) {
                ReflectionUtils.makeAccessible(f);
                ReflectionUtils.setField(f, adminHubConfig, null);                
            }
            
        }
    }
```

Reply:
@rjrudin Could we instead shift hubconfigimpl for the next minor release to be an extension of appconfig and resolve it in that effort?

Not saying we shouldn't use reflection atm, but just that the longer-term solution consist of a better way to handle configs? We'll esp have to start considering multi-users logged in for operational use (maybe not modifying the project, but just different users running flows/getting data, and right now that's tied at the hip with the hubconfig.)